ChemistryLib/PhreeqcInterfaceData/KineticReactant.h

```diff
@@ -0,0 +1,47 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2019, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#pragma once
+
+#include <boost/optional/optional.hpp>
+#include <fstream>
+#include <string>
+#include <vector>
+
+namespace BaseLib
+{
+class ConfigTree;
+}
+
+namespace ChemistryLib
+{
+struct KineticReactant
+{
+    KineticReactant(std::string name_,
+                    double amount_,
+                    boost::optional<std::vector<double>>
+                        parameters_)
+        : name(std::move(name_)),
+          amount(amount_),
+          parameters(parameters_ ? *parameters_ : std::vector<double>{})
+    {
+    }
+
+    friend std::ofstream& operator<<(
+        std::ofstream& out,
+        std::vector<KineticReactant> const& kinetic_reactants);
+
+    std::string const name;
+    double amount;
```

Comment:
const?

Reply:
Likely, in `std::ifstream& operator>>(std::ifstream& in, phreeqcIO& phreeqc_io)` (see ChemistryLib/PhreeqcInterface.cpp)
```
                case ItemType::KineticReactant:
                {
                    // Update amounts of kinetic reactants
                    auto& kinetic_reactant = BaseLib::findElementOrError(
                        kinetic_reactants.begin(), kinetic_reactants.end(),
                        [&item_name](KineticReactant const& kinetic_reactant) {
                            return kinetic_reactant.name == item_name;
                        },
                        "Could not find kinetic reactant '" + item_name + "'.");
                    kinetic_reactant.amount = accepted_items[item_id];
                    break;
                }
```
the amount of a kinetic reactant has to be updated as well.